### PR TITLE
Remove stack fault on longer running jobs

### LINKED ISSUE #971: [Bug] z/OS job submit does not seem to honor wait time and results in stack trace.
### Is there an existing issue for this?

- [X] There are no existing issues.

### Are the dependencies a supported version?

- [X] The dependencies are supported.

### IBM Z Open Automation Utilities

v1.2.5

### IBM Enterprise Python

v3.14.x

### IBM z/OS Ansible core Version

v1.5.0

### ansible-version

latest

### z/OS version

None

### Ansible module

_No response_

### Bug description

To execute the correct JCL the time of execution on None fail even before just throw the error 

JCL to check
```
JCL_FILE_CONTENTS_RC_8 = """//RCBADJCL JOB MSGCLASS=A,MSGLEVEL=(1,1),NOTIFY=&SYSUID,REGION=0M
//S1 EXEC PGM=IDCAMS
//SYSPRINT DD SYSOUT=*
//SYSIN DD *
 DELETE THIS.DATASET.DOES.NOT.EXIST
/*
"""
```

### Playbook verbosity output.

```
"ZOAU error":(0,"b""
{\"changed\": false, \"failed\": true, \"msg\": \"The JCL submitted with job id JOB00203 but there was an error, please review the error for further details: [ZOAUResponse]
	rc: 3
	response_format: UTF-8
	stdout_response: 
	stderr_response: BGYSC5403E Job is in phase STAT_SELECT: Awaiting selection on main and is not ready for DD queries.

	command: ddls JOB00203 ""\", \"invocation\": {\"module_args\": {\"src\": \"/tmp/tmp1vw2l37_\", \"location\": \"LOCAL\", \"max_rc\": null, \"wait_time_s\": 10, \"temp_file\": \"/tmp/ansible.p2guxm5l\", \"wait\": false, \"return_output\": true, \"use_template\": false, \"encoding\": null, \"volume\": null, \"template_parameters\": null, \"from_encoding\": \"UTF-8\", \"to_encoding\": \"IBM-1047\"}}}
","b""")
```

### Ansible configuration.

_No response_

### Contents of the inventory

_No response_

### Contents of `group_vars` or `host_vars`

_No response_

### PR DESCRIPTION
**Pull request by richp405**
##### SUMMARY
Minor changes to: 

      1. broaden exception handling in job.py
      2. Un-hide a duration issue that has been corrected.

##### FIXES
Fixes #971 

##### PULL TYPE
Bugfix Pull Request

##### COMPONENTS
job.py
test_job_submit_func.py

### CODE REVIEW

**Comment by richp405** [comment]
image

**Review by fernandofloresg — APPROVED** [review]

File: plugins/module_utils/job.py [hunk 1]
```diff
@@ -316,9 +316,8 @@ def _get_job_status(job_id="*", owner="*", job_name="*", dd_name=None, dd_scan=T
                 try:
                     list_of_dds = jobs.list_dds(entry.job_id)
                 except exceptions.DDQueryException as err:
-                    if 'BGYSC5201E' in str(err):
-                        is_dd_query_exception = True
-                        pass
+                    is_dd_query_exception = True
+                    pass
```
Comment (rexemin) [inline plugins/module_utils/job.py:320]: The `pass` keyword is not needed since we have another line running inside this block

**Review by rexemin — CHANGES_REQUESTED** [review]
Looks good, just left one comment

**Comment by fernandofloresg** [comment]
Ran the full test for this module and others and everything is fine 
not sure why log fragment check keeps erroring...
Screenshot 2024-04-22 at 2 05 11 PM

File: plugins/module_utils/job.py (hunk 1 above)
Comment (fernandofloresg) [reply plugins/module_utils/job.py]: updated it please review

**Review by rexemin — APPROVED** [review]

**Pull request merged**